shell/components/nav/WindowManager/ContainerShell.vue

```diff
@@ -121,14 +138,31 @@ export default {
     width() {
       this.fit();
     },
+
+    isXtermContainerFocused: {
+      handler(neu) {
+        const shellEl = document.querySelector(`.${ XTERM_SHELL_CLASS }`);
+
+        if (shellEl) {
+          shellEl.tabIndex = neu ? -1 : 0;
+        }
+      },
+      immediate: true
+    }
   },
 
   beforeUnmount() {
+    document.removeEventListener('keyup', this.handleKeyPress);
+    document.removeEventListener('focusin', this.focusChanged);
+
     clearInterval(this.keepAliveTimer);
     this.cleanup();
   },
 
   async mounted() {
+    document.addEventListener('keyup', this.handleKeyPress);
+    document.addEventListener('focusin', this.focusChanged);
```

Comment:
Try to avoid adding event listeners on the window and document objects. As we've seen in #13375, this practice has introduced conflicts with keyboard shortcuts that can be difficult to troubleshoot and resolve.

Reply:
Fixed